### PR TITLE
Proxy

### PR DESCRIPTION
**Pull request by romainAA**
#1232 
Should work, but I'm not sure how to test that it is indeed "piping" de data

### CODE REVIEW

**Review by houshuang — CHANGES_REQUESTED** [review]
Great that you've solved this problem, but could you please check your diffs? Why did you need to do a search and replace from request to req in this entire file? Can you redo the changes so that you are only adding the actual code for the proxy? 

Also, are you sure you need the Meteor http package?

**Comment by romainAA** [comment]
@houshuang done cleaning

**Comment by houshuang** [comment]
I get this error when I try http://localhost:3000/api/proxy?url=http://reganmian.net/blog

```js
W20180820-09:44:22.269(2)? (STDERR) (node:11864) UnhandledPromiseRejectionWarning: Error: options.uri is a required argument
W20180820-09:44:22.270(2)? (STDERR)     at Request.init (/Users/stian/src/frog/node_modules/request/request.js:231:31)
W20180820-09:44:22.270(2)? (STDERR)     at new Request (/Users/stian/src/frog/node_modules/request/request.js:127:8)
W20180820-09:44:22.270(2)? (STDERR)     at request (/Users/stian/src/frog/node_modules/request/index.js:53:10)
W20180820-09:44:22.270(2)? (STDERR)     at WebApp.connectHandlers.use (server/api.js:113:84)
W20180820-09:44:22.271(2)? (STDERR)     at /Users/stian/.meteor/packages/promise/.0.11.1.1044ql1.idgd++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
W20180820-09:44:22.271(2)? (STDERR) (node:11864) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
W20180820-09:44:22.271(2)? (STDERR) (node:11864) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
=> Finished delayed build of web.browser.legacy in 24695ms
```

File: frog/server/api.js [hunk 1]
```diff
@@ -99,6 +100,18 @@ const safeDecode = (query, field, msg, response, returnUndef) => {
 
 const InstanceDone = {};
 
+WebApp.connectHandlers.use('/api/proxy', (request, response, next) =>
+  request
+    .pipe(
+      requestFun(
+        require('url')
```
Comment (houshuang) [inline frog/server/api.js]: why do you require url here? why not import it at the top of the file?

**Review by houshuang — CHANGES_REQUESTED** [review]

File: frog/server/api.js (hunk 1 above)
Comment (houshuang) [reply frog/server/api.js]: also see the error

**Comment by romainAA** [comment]
For your error, in the issue you wrote that the url should be like this:
http://localhost:3000/api/proxy/https://github.com
whereas yours is like:
http://localhost:3000/api/proxy?url=https://github.com
Which one should it be ?

**Pull request merged**